core/modules/comment/comment.module

```diff
@@ -67,6 +67,22 @@ define('COMMENT_NODE_CLOSED', 1);
  */
 define('COMMENT_NODE_OPEN', 2);
 
+/**
+ * Auto generated comment titles.
+ */
+define('COMMENT_TITLE_AUTO', 0);
+
+/**
+ * Custom comment titles.
+ */
+define('COMMENT_TITLE_CUSTOM', 1);
+
+/**
+ * Hidden comment titles.
+ */
+define('COMMENT_TITLE_HIDDEN', 2);
```

Comment:
If we are fixed to using constants for these options, can we please not map them to "arbitrary" numbers? This causes things to be saved in config like `"comment_title_options": "2"` - I'd rather this was easy to read and self-explanatory instead, like `"comment_title_options": "hidden"` ...I think I would actually prefer it if it was something like `"comment_title_display": "hidden"`

Reply:
I'd like another opinion on this as I was originally advised to do it the other way and am not qualified to say which is better.

Reply:
It is not an unreasonable suggestion, as @klonos said this would make reading config files easier. But its not wrong the way it is either, as we have a mix of constant mappings in core it seems: sometimes we use integers (eg `define('LAYOUT_STORAGE_DEFAULT', 4)` and `define('NODE_PUBLISHED', 1)`), other times strings (eg `define('LAYOUT_TITLE_CUSTOM', 'custom')` and `define('NODE_ACCESS_ALLOW', 'allow')`). Again this could be another issue/initiative for switching to more readable config files but shouldnt block this.

Reply:
@docwilmot agreed re the fact that we have either been adding or have inherited many such "arbitrary" values that make people scratch their heads when they see them. Also agreed that we could leave things as they are with the current PR in order to not block it, and then file a follow-up to clean things up 👍🏼 ...but since we have time till feature freeze, I say why add to the already big, messy pile of things we will need to fix in the future, when we have the opportunity to polish this PR now?

my 2c

Reply:
OK, I implemented these changes. We'll see if anyone else has anything to add.